Add recursive directive for deep nested nodes

m1llen1um opened the pull request:
#836 ng-include trigger digest and with trees 10 lvl deep 10 digest appears. An solution is to use [recursive directive](https://github.com/marklagendijk/angular-recursion) with cached template. Simple solution which do not change current architecture, but just adds:

1 attribute directive - **ui-tree-nodes-renderer**
2 directive attributes - **nodeRenderTemplate**, **currentNode**

notes: _no tests are applied; recommendations and changes to my solution are wellcome_

Voles commented:
@m1llen1um first of all, thanks for your effort! Can you please check your PR with our [Contributing Guidelines](https://github.com/angular-ui-tree/angular-ui-tree/blob/master/CONTRIBUTING.md)?

Then we can start the review process. Thanks in advance!

m1llen1um commented:
@Voles I changed the PR title and removed the dist/ changes. I hope everything is ok now. If you find something not corresponding to your _"Contributing Guidelines"_ please point to it.

dayanrr91 commented:
the issue is still happening, with 9 levels I have an error
